packages/checkout-storefront/src/hooks/useCheckoutUpdateStateTrigger.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  CheckoutFormData,
+  CheckoutUpdateStateScope,
+} from "@/checkout-storefront/sections/CheckoutForm/types";
+import { useEffect } from "react";
+import { useFormContext } from "react-hook-form";
+
+export const useCheckoutUpdateStateTrigger = (
+  scope: CheckoutUpdateStateScope,
+  updating: boolean
+) => {
+  const { setValue } = useFormContext<CheckoutFormData>();
+
+  useEffect(() => {
+    setValue(`updateState.${scope}`, updating);
```

Comment:
Why are we setting some values in `react-hook-form`? Is it just to trigger some validation updates?

Reply:
Yeah, it's because checkout finalization now needs to wait till all api changes happen, hence we're have a state in checkout form so it knows whether to wait or not